### ISSUE TITLE
[BUG]: SqlAzureDacpacDeployment@1 - error when using a Workload Identity federation

### ISSUE THREAD

**Issue by zoemac11** [microsoft/azure-pipelines-tasks#19721]
### New issue checklist

- [X] I searched for [existing GitHub issues](https://github.com/microsoft/azure-pipelines-tasks/issues)
- [X] I read [pipeline troubleshooting guide](https://docs.microsoft.com/vsts/build-release/actions/troubleshooting)
- [X] I checked [how to collect logs](https://learn.microsoft.com/azure/devops/pipelines/troubleshooting/review-logs?view=azure-devops)

### Task name

SqlAzureDacpacDeployment@1

### Task version

1.237.4

### Issue Description

Getting the following error when using a Workload Identity federation service connection

`##[error]Microsoft.PowerShell.Commands.WriteErrorException: ExceptionMessage: Cannot bind argument to parameter 'vstsAccessToken' because it is an empty string. (in function: Build-MSALInstance)Check out how to troubleshoot failures at https://aka.ms/sqlazuredeployreadme#troubleshooting-`

### Environment type (Please select at least one enviroment where you face this issue)

- [ ] Self-Hosted
- [X] Microsoft Hosted
- [ ] VMSS Pool
- [ ] Container

### Azure DevOps Server type

dev.azure.com (formerly visualstudio.com)

### Operation system

Microsoft Windows Server 2022

### Relevant log output

```shell
##[error]Microsoft.PowerShell.Commands.WriteErrorException: ExceptionMessage: Cannot bind argument to parameter 'vstsAccessToken' because it is an empty string. (in function: Build-MSALInstance)Check out how to troubleshoot failures at https://aka.ms/sqlazuredeployreadme#troubleshooting-
```

### Full task logs with system.debug enabled

<details>
  <pre> Starting: Run migrations
==============================================================================
Task         : Azure SQL Database deployment
Description  : Deploy an Azure SQL Database using DACPAC or run scripts using SQLCMD
Version      : 1.237.4
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/deploy/sql-azure-dacpac-deployment
==============================================================================
Added TLS 1.2 in session.
Temporary inline SQL file: C:\Users\VssAdministrator\AppData\Local\Temp\tmp56C8.tmp
Invoke-Sqlcmd -connectionString "**********"  -Inputfile "C:\Users\VssAdministrator\AppData\Local\Temp\tmp56C8.tmp" 
##[error]Microsoft.PowerShell.Commands.WriteErrorException: ExceptionMessage: Cannot bind argument to parameter 'vstsAccessToken' because it is an empty string. (in function: Build-MSALInstance)Check out how to troubleshoot failures at https://aka.ms/sqlazuredeployreadme#troubleshooting-
Finishing: Run migrations
 </pre>
</details>

### Repro steps

```yml
pool: 
    vmImage: 'windows-latest'

  steps:           
    - task: SqlAzureDacpacDeployment@1
      displayName: 'Run migrations'
      inputs:
        azureSubscription: '<workload-identity-service-connection>'
        AuthenticationType: connectionString
        ConnectionString: 'Server=database.database.windows.net;Database=database;User ID=user;Password=password;'
        deployType: 'SqlTask'
        SqlFile: '$(System.DefaultWorkingDirectory)/migrations/migrations.sql'
        IpDetectionMethod: 'AutoDetect'
```

**Comment by verhaje** [comment]
Same configuration, same error. Any update on the cause or solution?

**Comment by D0nnyg** [comment]
For me the fix was to enable the following Azure SQL Database Networking option:

<img width="384" alt="image" src="https://github.com/microsoft/azure-pipelines-tasks/assets/101480881/2f59bb6d-da94-4d3e-986c-cfa929e2974c">

It looks like the underlying Azure DevOps server cannot reach the Azure Database.

The required steps:
- Navigate to https://portal.azure.com/
- Go to the (Subscription and then) Resource Group containing the Azure SQL Database
- Go to the Azure SQL Database component
- Go to "Security/Networking" in the pane on the left side
- If you have Public Access enabled for "Selected Networks" the above shown option is at the bottom of the screen

Hope this helps

**Comment by tomqiaozc** [comment]
I encountered the same problem. The service connection established by my pipeline using the Client secrets method in the Service principal was no problem. But now Client secrets are deprecated, so I tried managed Identity and Federated credentials in Service principal respectively. After the service connection of managed Identity was created, I successfully ran the sql DB task, but the problem occurred when I tried to rerun it again today and it failed.
My agent pool is Windows 2019.
